gpMgmt/test/behave/mgmt_utils/steps/recoverseg_mgmt_utils.py

```
@@ -754,3 +754,85 @@ def get_host_address(hostname):
+        search_ip_addr = context.host_ip_list[host]
+        dbname = "template1"
+        ip_address = tuple(search_ip_addr) if len(search_ip_addr) > 1 else "('{}')".format(search_ip_addr[0])
+        query = "SELECT count(*) FROM pg_hba_file_rules WHERE database='{{replication}}' AND (address='{0}' OR address IN {1})".format(
+            host, ip_address)
+        phost = seg.primaryDB.getSegmentHostName()
+        port = seg.primaryDB.getSegmentPort()
+        print(query)
+        with closing(dbconn.connect(dbconn.DbURL(dbname=dbname, port=port, hostname=phost),
+                                    utility=True, unsetSearchPath=False)) as conn:
+            result = dbconn.querySingleton(conn, query)
+            if result != 0:
+                raise Exception("{0} replication entry for {1}, {2} still existing in pg_hba.conf of {3}:{4}"
+                                .format(result, host, search_ip_addr,phost, port))
+
+
+@then('verify that only replication connection primary has is to {new_mirror}')
+@when('verify that only replication connection primary has is to {new_mirror}')
+@given('verify that only replication connection primary has is to {new_mirror}')
+def impl(context, new_mirror):
+    all_segments = GpArray.initFromCatalog(dbconn.DbURL()).getSegmentList()
+
+    for seg in all_segments:
+        if seg.mirrorDB.getSegmentHostName() != new_mirror:
+            continue
+
+        dbname = "template1"
+        search_ip_addr = context.host_ip_list[new_mirror]
+        ip_address = tuple(search_ip_addr) if len(search_ip_addr) > 1 else "('{}')".format(search_ip_addr[0])
+        query = """
+        SELECT 'old_host'  AS condition, COUNT(*) AS count
+        FROM pg_catalog.gp_stat_replication
+        WHERE client_addr NOT IN {0}
+        UNION
+        SELECT 'new_host' AS condition, COUNT(*) AS count
+        FROM pg_catalog.gp_stat_replication where client_addr IN {1}
+        """.format(ip_address, ip_address)
```

I think we can achieve the same result using this SQL - 
```
SELECT
  CASE
    WHEN
      (SELECT COUNT(*) FROM gp_stat_replication WHERE client_addr IN (<address>)) =
      (SELECT COUNT(*) FROM gp_stat_replication)
    THEN TRUE
    ELSE FALSE
  END;
```
We are querying in utility mode, the results returned by the SQL with the `client_addr` filter should equal all the rows present in the table, since each segment will have only a single replication. If they are not identical, then there are some other replication connections that are not present in the `<address>` field which should throw an error.